modules/azure-gaming/game-dev-vmss/main.bicep

```diff
@@ -0,0 +1,247 @@
+
+@description('File Share Storage Account name')
+param fileShareStorageAccount string = ''
+
+@description('File Share Storage Account key')
+@secure()
+param fileShareStorageAccountKey string = ''
+
+@description('File Share name')
+param fileShareName string = ''
+
+@description('Perforce Port address')
+param p4Port string = ''
+
+@description('Perforce User')
+param p4Username string = ''
+
+@description('Perforce User password')
+@secure()
+param p4Password string = ''
+
+@description('Perforce Client Workspace')
+param p4Workspace string = ''
+
+@description('Perforce Stream')
+param p4Stream string = ''
+
+@description('Perforce Depot Client View mappings')
+param p4ClientViews string = ''
+
+@description('Incredibuild License Key')
+@secure()
+param ibLicenseKey string = ''
+
+@description('GDK Version')
+param gdkVersion string = 'June_2021_Update_4'
+
+@description('Use VM to sysprep an image from')
+param useVmToSysprepCustomImage bool = false
+
+@description('Remote Access technology')
+@allowed([
+  'RDP'
+  'Teradici'
+  'Parsec'
+])
+param remoteAccessTechnology string = 'RDP'
+
+@description('Teradici Registration Key')
+@secure()
+param teradiciRegKey string = ''
+
+@description('Parsec Team ID')
+param parsecTeamId string = ''
+
+@description('Parsec Team Key')
+@secure()
+param parsecTeamKey string = ''
+
+@description('Parsec Hostname')
+param parsecHost string = ''
+
+@description('Parsec User Email')
+param parsecUserEmail string = ''
+
+@description('Parsec Is Guest Access')
+param parsecIsGuestAccess bool = false
+
+@description('Virtual Network Resource Name')
+param vnetName string = 'vnet-${vmssName}'
+
+@description('Virtual Network Subnet Name')
+param subnetName string = 'subnet${vmssName}'
+
+@description('Virtual Network Security Group Name')
+param networkSecurityGroupName string = 'nsg-${vmssName}'
+
+@description('Virtual Network Address Prefix')
+param vnetAddressPrefix string = '172.17.72.0/24' //Change as needed
+
+@description('Virtual Network Subnet Address Prefix')
+param subnetAddressPrefix string = '172.17.72.0/25' // 172.17.72.[0-128] is part of this subnet
+
+var customData = format('''
+fileShareStorageAccount={0}
+fileShareStorageAccountKey={1}
+fileShareName={2}
+p4Port={3}
+p4Username={4}
+p4Password={5}
+p4Workspace={6}
+p4Stream={7}
+p4ClientViews={8}
+ibLicenseKey={9}
+gdkVersion={10}
+useVmToSysprepCustomImage={11}
+remoteAccessTechnology={12}
+teradiciRegKey={13}
+parsecTeamId={14}
+parsecTeamKey={15}
+parsecHost={16}
+parsecUserEmail={17}
+parsecIsGuestAccess={18}
+deployedFromSolutionTemplate={19}
+''', fileShareStorageAccount, fileShareStorageAccountKey, fileShareName, p4Port, p4Username, p4Password, p4Workspace, p4Stream, p4ClientViews, ibLicenseKey, gdkVersion, useVmToSysprepCustomImage, remoteAccessTechnology, teradiciRegKey, parsecTeamId, parsecTeamKey, parsecHost, parsecUserEmail, parsecIsGuestAccess, false)
+
+
+module vnet './nestedtemplates/virtualNetworks.bicep'  = {
```

Comment:
have you looked into using the `network/virtual-network` module? If you cannot use it, can you share the feedback why? We'd prefer not to duplicate this.

Reply:
let me review #128 and see if that solves the issue I was having

Reply:
i have this PR where I was already trying to convert to using the module. Let me give it a few more tries now and see if I can leverage the module. (Otherwise, this PR will repro why I can not)

Reply:
cc @MrMCake as FYI

Reply:
@alex-frankel , @MrMCake , over in the quickstarts repo, I ran into a few best practice issues, which I detailed in #132

Reply:
I am also getting stuck here with the subnet.

Unable to edit or replace deployment 'subnetp3qxqfi6ccpbq': previous deployment from '8/3/2022 7:48:21 PM' is still active (expiration time is '8/10/2022 7:48:18 PM').

Reply:
@alex-frankel , I am going to give up on the vnet module within this first PR. I am starting a 2nd PR here, #134 where we can dive into exactly what issues are going on.